src/golang/lib/workflow/utils/utils.go

```diff
@@ -60,6 +54,16 @@ func ReadFromStorage(ctx context.Context, storageConfig *shared.StorageConfig, p
 	return nil
 }
 
+// Only to be used when duplicating cached results in the preview artifact cache.
+func CopyPathInStorage(ctx context.Context, storageConfig *shared.StorageConfig, fromPath string, toPath string) error {
+	s := storage.NewStorage(storageConfig)
+	content, err := s.Get(ctx, fromPath)
+	if err != nil {
+		return err
```

Comment:
So I believe that a race-with-eviction here is quite difficult to achieve, since a cache hit would have updated the entry's last used timestamp in the cache. And when this error is hit, it doesn't kill the workflow, but instead the operator will then try and execute without the cache.